### ISSUE TITLE
Feature Request: Multisig support

### ISSUE THREAD

**Issue by zpetersen-paxos** [solana-foundation/anchor#3134]
Hello, it looks like anchor does not support the token program multisig as described [here](https://spl.solana.com/token#multisig-usage).  I see there are some TODOs in the [code](https://github.com/coral-xyz/anchor/blob/fb7ea68c0de780c5be688b5dadad6f25dd4c45bc/spl/src/token.rs#L318C17-L318C22) that are very old.  

Is there a workaround to support this?  If not can this feature be added?

**Comment by acheroncrypto** [comment]
Correct, the wrappers in `anchor-spl` don't support `spl-token` multisigs, but you should be able to just copy the functions you'd like to use from the file you've linked and specify the signers array as a workaround.

**Comment by zpetersen-paxos** [comment]
Yea I did that and it's working.  To use this in any production grade code though we'd want to avoid this workaround.

I think it wouldn't be too difficult to add, just need a new field in the struct and then reference that.  Wdyt?

**Comment by acheroncrypto** [comment]
I don't know what the initial reasoning was for not supporting multisigs, but adding a new field for all these wrapper instructions would break pretty much everyone's code.

Given most people don't or won't use this feature (I don't think anyone asked about it before even though these functions are quite popular), it would be better to find a way to add this functionality without causing breakage.

**Comment by zpetersen-paxos** [comment]
Ah right struct fields in rust are required.  That definitely makes the backwards compatibility a bit trickier.  Any suggestions on how this could be achieved?
